example/reduce/src/reduce.cpp

```diff
@@ -0,0 +1,155 @@
+
+    if (blockCount > maxBlockCount)
+        blockCount = maxBlockCount;
+
+    // allocate memory
+    auto hostMemory = alpaka::mem::view::ViewPlainPtr<DevHost, T, Dim, Idx>(
+        new T[n], devHost, n);
+    alpaka::mem::buf::Buf<DevAcc, T, Dim, Extent> sourceDeviceMemory =
+        alpaka::mem::buf::alloc<T, Idx>(devAcc, n);
+
+    alpaka::mem::buf::Buf<DevAcc, T, Dim, Extent> destinationDeviceMemory =
+        alpaka::mem::buf::alloc<T, Idx>(
+            devAcc, static_cast<Extent>(blockCount));
+
+    T *nativeHostMemory = alpaka::mem::view::getPtrNative(hostMemory);
+
+    // fill array with data
+    for (uint32_t i = 0; i < n; i++)
+        nativeHostMemory[i] = static_cast<T>(i + 1);
+
+    // copy the data to the GPU
+    alpaka::mem::view::copy(queue, sourceDeviceMemory, hostMemory, n);
+
+    // define the reduction function
+    auto addFn = [] ALPAKA_FN_ACC(T a, T b) -> T { return a + b; };
+
+    // create kernels with their workdivs
+    ReduceKernel<blockSize, T, decltype(addFn)> kernel1, kernel2;
+    WorkDiv workDiv1{ static_cast<Extent>(blockCount),
+                      static_cast<Extent>(blockSize),
+                      static_cast<Extent>(1) };
+    WorkDiv workDiv2{ static_cast<Extent>(1),
+                      static_cast<Extent>(blockSize),
+                      static_cast<Extent>(1) };
+
+    // execute first kernel
+    auto const exec1(alpaka::kernel::createTaskExec<Acc>(
+        workDiv1,
+        kernel1,
+        alpaka::mem::view::getPtrNative(sourceDeviceMemory),
+        alpaka::mem::view::getPtrNative(destinationDeviceMemory),
+        n,
+        addFn));
+
+    // reduce the last block
+    auto const exec2(alpaka::kernel::createTaskExec<Acc>(
+        workDiv2,
+        kernel2,
+        alpaka::mem::view::getPtrNative(destinationDeviceMemory),
+        alpaka::mem::view::getPtrNative(destinationDeviceMemory),
+        blockCount,
+        addFn));
+
+    // enqueue both kernels
+    alpaka::queue::enqueue(queue, exec1);
+    alpaka::queue::enqueue(queue, exec2);
+
+    //  download result from GPU
+    T resultGpuHost;
+    auto resultGpuDevice =
+        alpaka::mem::view::ViewPlainPtr<DevHost, T, Dim, Idx>(
+            &resultGpuHost, devHost, (Extent)blockSize);
+
+    alpaka::mem::view::copy(queue, resultGpuDevice, destinationDeviceMemory, 1);
+
+    // check result
+    if (resultGpuHost != static_cast<T>(n / 2 * (n + 1)))
+    {
+        std::cout << "Results don't match: " << resultGpuHost << " != " << n
+                  << "\n";
+        return false;
+    }
+    else
+    {
+        std::cout << "Results match.\n";
+        return true;
+    }
+}
+
+int main()
+{
+    // select device and problem size
+    const int dev = 0;
+    uint64_t n = 1 << 28;
+
+    DevAcc devAcc(alpaka::pltf::getDevByIdx<PltfAcc>(dev));
+
+    // reduce
+    bool result = reduce(devAcc, n);
+
+    // clean up
+    alpaka::dev::reset(devAcc);
+
+    return (result ? 0 : -1);
```

Comment:
Please use `EXIT_SUCCESS` and `EXIT_FAILURE`.

Reply:
note: include `<cstdlib>` for that https://en.cppreference.com/w/cpp/utility/program/EXIT_status